ec2/security.go

```diff
@@ -0,0 +1,77 @@
+package ec2
+
+import (
+	"context"
+	"github.com/YaleSpinup/apierror"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	log "github.com/sirupsen/logrus"
+)
+
+// ListSecurityGroups List all security groups in an aws account
+func (e *Ec2) ListSecurityGroups(ctx context.Context, org string) ([]map[string]*string, error) {
+	log.Infof("listing ec2 security groups (org: '%s')", org)
+
+	var filters []*ec2.Filter
+	if org != "" {
+		filters = []*ec2.Filter{inOrg(org)}
+	}
+
+	input := ec2.DescribeSecurityGroupsInput{
+		GroupIds: aws.StringSlice([]string{}),
+		Filters:  filters,
+	}
+
+	out, err := e.Service.DescribeSecurityGroupsWithContext(ctx, &input)
+	if err != nil {
+		return nil, ErrCode("listing security groups", err)
+	}
+
+	log.Debugf("returning list of %d snapshots", len(out.SecurityGroups))
+
+	list := make([]map[string]*string, len(out.SecurityGroups))
+	for i, s := range out.SecurityGroups {
+		tags := s.Tags
+		var sgName *string
+
+		// Loop through the tags and if Name exist, set the sgName value to it
+		for _, t := range tags {
+			if *t.Key == "Name" {
+				sgName = t.Value
+			}
+		}
+
+		// If sgName is nil, use the GroupName on the security group as a fallback
+		if sgName == nil {
+			sgName = s.GroupName
+		}
+
+		list[i] = map[string]*string{
+			*s.GroupId: sgName,
+		}
```

Comment:
I would suggest a couple of things here:
- set the default value for `sgName` first and then just override it
- don't bother with the `tags` variable, its unecessary
- use the aws sdk helper utilities
- break out of the loop once you find the Name tag

something like this:

```golang
	for i, s := range out.SecurityGroups {
		sgName := s.GroupName
		for _, t := range s.Tags {
			if aws.StringValue(t.Key) == "Name" {
				sgName = t.Value
				break
			}
		}

		list[i] = map[string]*string{
			aws.StringValue(s.GroupId): sgName,
		}
	}
```